Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -895,6 +895,7 @@ internal struct BinaryDecoder: Decoder {
 
         if let k = k, let v = v {
             value[k] = v
+            consumed = true
```

Comment:
So in addition to the test I tagged in the second commit in this PR putting it any other place than here causes some other things to fail.  I need to go back to the C++ code to figure out what happens if a map is missing the key or value on the wire.  As we're currently written, I don't think we'll parse it, but when I did maps for C++ I almost thing the C++ would end up using the default value.  The really crappy part is for a proto2 syntax file, the map can be to enums, so the value could be non zero making it harder to hardcode.